index.html

```diff
@@ -2866,32 +2836,57 @@ <h3><dfn>Writing or pushing content</dfn></h3>
         <a href="https://www.w3.org/2001/tag/doc/URNsAndRegistries-50.html">
         W3C TAG recommendations</a> to avoid using URNs in the Web.
         Therefore, Web NFC does not use the URN prefix neither when reading
-        or writing <a>NDEF messages</a>, but requires the leading colon ('`:`')
-        before the type name.
+        or writing <a>NDEF messages</a>.
       </p>
       <p>
-        To <dfn>validate external type</dfn> given a |input:USVString| run
+        To <dfn>validate external type</dfn> given |input:USVString|, run
         these steps:
       </p>
       <ol class=algorithm id="validate-external-type">
         <li>
           If |input:string| is not a {{USVString}} or it is empty, or its
-          length exceeds 255 bytes, or it contains white spaces,
-          return `false`.
+          length exceeds 255 bytes, return `false`.
         </li>
         <li>
-          If |input| does not start with colon ('`:`'), return |input|.
-          <p class="issue">
-            Validating domain name needs an algorithm.
-          </p>
+          Let  |colonIndex| be the last occurrence of `U+003A` (`:`) in |input|.
+          If a `U+003A` (`:`) is not found, return `false`.
```

Comment:
Nit: Shall we swap the order of those two sentences?

Reply:
I was thinking about that, but the way you write code, it's an error/exception handling. If I put it to the front, it's double pass if someone takes the steps literally. Maybe there is a better formulation?

Reply:
Or we could do like url spec:

> If bytes contains a 0x3D (=), then let name be the bytes from the start of bytes up to but excluding its first 0x3D (=), and let value be the bytes, if any, after the first 0x3D (=) up to the end of bytes. If 0x3D (=) is the first byte, then name will be the empty byte sequence. If it is the last, then value will be the empty byte sequence.

If |input| contains a `U+003A` (`:`), then let |domain| be the bytes from the start of bytes up to but excluding its first `U+003A` (`:`), and let |type| be the bytes, if any, after the first `U+003A` (`:`) up to the end of bytes.

Reply:
If |input| does not contains a U+003A (:), return false

Let |domain| be the |input| from the start of |input| up to but excluding its first U+003A (:), and let |type| be the |input|, if any, after the first U+003A (:) up to the end of |input|.

Reply:
except that it's the last occurrence of `U+003A`.

Reply:
And then need to check if |domain| and |type| are defined, i.e. in the steps above we need to include "or otherwise `null`

Reply:
Is it last though? We allow URLs but the scheme and port etc is not written down.

so we accept people to write "https://intel.com:8080:coolio" or actually "intel.com:coolio" - in the latter one there is only one : ever

Reply:
If |input| does not contains a U+003A (:), return false

Let |domain| be the |input| from the start of |input| up to but excluding its last U+003A (:), and let |type| be the |input|, if any, after the last U+003A (:) up to the end of |input|.

Reply:
This needs another round now. The sentence above doesn't handle the case when a colon is not found.